### PR TITLE
Adding Supernova keyboard to Charue family

### PR DESCRIPTION
**Pull request by PiKeeb**
## Description

Adding Supernova keyboard to Charue family:
https://charue-design.com/collections/supernova

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/charue/supernova/config.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+/* Keyboard Matrix Assignments */
+#define MATRIX_ROW_PINS { B3, B2, B1, E6, B7, B0 }
+#define MATRIX_COL_PINS { F0, F1, F4, F7, F6, F5, C7, C6, B6, B5, B4, D7, D6, D4, D5, D3 }
+#define DIODE_DIRECTION COL2ROW
+
+/* Encoder */
+#define ENCODERS_PAD_A { D0 }
+#define ENCODERS_PAD_B { D1 }
+#define ENCODER_RESOLUTION 4
+#define ENCODER_DIRECTION_FLIP
```
Comment (zvecr) [inline keyboards/charue/supernova/config.h:41]:
```suggestion
#define ENCODERS_PAD_A { D1 }
#define ENCODERS_PAD_B { D0 }
#define ENCODER_RESOLUTION 4
```

File: keyboards/charue/supernova/supernova.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code_delay(KC_VOLU, 10);
+        } else {
+            tap_code_delay(KC_VOLD, 10);
+        }
+    }
+    return encoder_update_user(index, clockwise);
```
Comment (zvecr) [inline keyboards/charue/supernova/supernova.c:27]:
```suggestion
    if (!encoder_update_user(index, clockwise)) return false;
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return false;
```

File: keyboards/charue/supernova/info.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
+{
+    "keyboard_name": "Supernova",
+    "url": "https://charue-design.com/collections/supernova",
+    "maintainer": "ILWrites",
+    "width": 16,
+    "height": 6,
```
Comment (zvecr) [inline keyboards/charue/supernova/info.json:6]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/charue/supernova/keymaps/debug/keymap.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
```
Comment (drashna) [inline keyboards/charue/supernova/keymaps/debug/keymap.c:29]:
```suggestion
```

File: keyboards/charue/supernova/config.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+/* Keyboard Matrix Assignments */
+#define MATRIX_ROW_PINS { B3, B2, B1, E6, B7, B0 }
+#define MATRIX_COL_PINS { F0, F1, F4, F7, F6, F5, C7, C6, B6, B5, B4, D7, D6, D4, D5, D3 }
+#define DIODE_DIRECTION COL2ROW
+
+/* Encoder */
+#define ENCODERS_PAD_A { D1 }
+#define ENCODERS_PAD_B { D0 }
+#define ENCODER_RESOLUTION 4
```
Comment (drashna) [inline keyboards/charue/supernova/config.h:40]:
```suggestion
```

File: keyboards/charue/supernova/keymaps/debug/keymap.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+    };
+    return true;
+};
```
Comment (drashna) [inline keyboards/charue/supernova/keymaps/debug/keymap.c:70]:
```suggestion
```

File: keyboards/charue/supernova/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+# Build Options
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/charue/supernova/rules.mk:16]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/charue/supernova/supernova.c [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "supernova.h"
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/charue/supernova/supernova.c:19]:
```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
```

File: keyboards/charue/supernova/supernova.c [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    if (!encoder_update_user(index, clockwise)) return false;
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code_delay(KC_VOLU, 10);
+        } else {
+            tap_code_delay(KC_VOLD, 10);
+        }
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/charue/supernova/supernova.c:29]:
```suggestion
}
#endif

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/charue/supernova/keymaps/debug/keymap.c (hunk 6 above)
Comment (PiKeeb) [reply keyboards/charue/supernova/keymaps/debug/keymap.c:64]: I am aware that these are examples of macros. However, this is a debug keymap. It's important to check the macro functionality as well. They are not present in the default or via keymaps.

**Pull request closed without merging**